helmclient.go

```diff
@@ -698,7 +700,7 @@ func (c *Client) newTunnel() (*k8sportforward.Tunnel, error) {
 	err = validateTillerVersion(pod, c.tillerImage)
 	if err != nil {
 		if IsTillerInvalidVersion(err) && !c.tillerUpgradeEnabled {
-			c.logger.Log("level", "debug", "message", "found an out-dated tiller but keep going to create a tunnel")
+			return nil, microerror.Maskf(tillerOutdatedError, "found an out-dated tiller but upgrades not enabled")
```

Comment:
Tiller is upgraded by chart-operator in the tenant cluster. When we want to upgrade Tiller we deploy a new version of chart-operator.

So when upgrades are disabled in app-operator we should return the error and cancel the resource.